test/browser/fixtures/expect-dom/toContainHTML.test.ts

```diff
@@ -0,0 +1,120 @@
+    expect(parent).toContainHTML(incorrectStringHtml)
+    expect(child).toContainHTML(incorrectStringHtml)
+
+    // negative test cases wrapped in throwError assertions for coverage.
+    expect(() =>
+      expect(nonExistantElement).not.toContainHTML(stringChildElement),
+    ).toThrowError()
+    expect(() =>
+      // @ts-expect-error testing invalid input
+      expect(nonExistantElement).not.toContainHTML(nonExistantElement),
+    ).toThrowError()
+    expect(() =>
+      // @ts-expect-error testing invalid input
+      expect(stringChildElement).not.toContainHTML(fakeElement),
+    ).toThrowError()
+    expect(() =>
+      expect(svgElement).toContainHTML(stringChildElement),
+    ).toThrowError()
+    expect(() =>
+      expect(grandparent).not.toContainHTML(stringChildElement),
+    ).toThrowError()
+    expect(() =>
+      expect(parent).not.toContainHTML(stringChildElement),
+    ).toThrowError()
+    expect(() =>
+      expect(child).not.toContainHTML(stringChildElement),
+    ).toThrowError()
+    expect(() =>
+      expect(child).not.toContainHTML(stringChildElement),
+    ).toThrowError()
+    expect(() =>
+      expect(child).not.toContainHTML(stringChildElementSelfClosing),
+    ).toThrowError()
+    expect(() => expect(child).toContainHTML(nonExistantString)).toThrowError()
+    expect(() => expect(parent).toContainHTML(nonExistantString)).toThrowError()
+    expect(() =>
+      expect(grandparent).toContainHTML(nonExistantString),
+    ).toThrowError()
+      // @ts-expect-error testing invalid input
+    expect(() => expect(child).toContainHTML(nonExistantElement)).toThrowError()
+    expect(() =>
+      // @ts-expect-error testing invalid input
+      expect(parent).toContainHTML(nonExistantElement),
+    ).toThrowError()
+    expect(() =>
+      // @ts-expect-error testing invalid input
+      expect(grandparent).toContainHTML(nonExistantElement),
+    ).toThrowError()
+    expect(() =>
+      expect(nonExistantElement).not.toContainHTML(incorrectStringHtml),
+    ).toThrowError()
+    expect(() =>
+      expect(grandparent).not.toContainHTML(incorrectStringHtml),
+    ).toThrowError()
+    expect(() =>
+      expect(child).not.toContainHTML(incorrectStringHtml),
+    ).toThrowError()
+    expect(() =>
+      expect(parent).not.toContainHTML(incorrectStringHtml),
+    ).toThrowError()
+  })
+
+  test('throws with an expected text', async () => {
+    const {queryByTestId} = render('<span data-testid="child"></span>')
+    const htmlElement = queryByTestId('child')
+    const nonExistantString = '<div> non-existant element </div>'
+
+    let errorMessage
+    try {
+      expect(htmlElement).toContainHTML(nonExistantString)
+    } catch (error) {
+      errorMessage = error.message
+    }
+
+    expect(errorMessage).toMatchInlineSnapshot(`
+      "expect(element).toContainHTML()
+      Expected:
+        <div> non-existant element </div>
+      Received:
+        <span
+        data-testid="child"
+      />"
```

Comment:
When running `BROWSER=chromium pnpm -C test/browser test-expect-dom` locally, I get diff due to color. Can we strip color before the assertion?

![image](https://github.com/user-attachments/assets/821f033e-6b39-40f9-ae52-081eb5b52564)